src/flags.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+use binding;
+use libc::c_int;
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::mem;
+
+// Creates vector of strings, Vec<String>
+#[cfg(test)]
+macro_rules! svec {
+    ($($x:expr),*) => (vec![$($x.to_string()),*]);
+}
+
+#[derive(Debug, PartialEq)]
+pub struct DenoFlags {
+  pub help: bool,
+  pub log_debug: bool,
+  pub version: bool,
+  pub reload: bool,
+  pub allow_write: bool,
+  pub allow_net: bool,
+}
+
+pub fn print_usage() {
+  println!(
+    "Usage: deno script.js
+
+-help         Print this message.
+-version      Print the version.
+-reload       Reload cached remote resources.
+-allow-write  Allow file system write access.
+-allow-net    Allow network access.
+-DEBUG        Log debug output."
+  );
+}
+
+// Parses flags for deno. This does not do v8_set_flags() - call that separately.
+pub fn set_flags(args: Vec<String>) -> (DenoFlags, Vec<String>) {
+  let mut flags = DenoFlags {
+    help: false,
+    version: false,
+    reload: false,
+    log_debug: false,
+    allow_write: false,
+    allow_net: false,
+  };
+  let mut rest = Vec::new();
+  for a in &args {
+    match a.as_str() {
+      "-h" | "-help" | "--help" => flags.help = true,
```

Comment:
I'm not sure you intended these flags as definitive, but in case you did...
* I've just been using powershell and it does flags in the for form `-flag`. However it seems more the exception than the rule, I'd suggest to stick to `-f and --flag` only.